UploadDaemon/Upload/FileSystemUpload.cs

```diff
@@ -48,6 +49,15 @@ public Task<bool> UploadAsync(string filePath, string version)
             }
         }
 
+        private void EnsureTargetDirectoryExists(string targetDirectory)
+        {
+            if (!Directory.Exists(targetDirectory))
+            {
+                logger.Debug("Creating target directory: {targetDirectory}", targetDirectory);
+                Directory.CreateDirectory(targetDirectory);
+            }
+        }
```

Comment:
Do we also need some handling if the directory path is already a file path as we have in the profiler itself?

Reply:
Good point.